raw_linux.go

```diff
@@ -279,6 +279,34 @@ func (p *packetConn) SetBPF(filter []bpf.RawInstruction) error {
 	)
 }
 
+// SetPromisc enables/disables interface promiscuous mode through syscall
+// when thread exits interface returns to previous state
+// 0 == disabled, 1 == enabled
+type packet_mreq struct {
```

Comment:
Awesome, I've updated the internals of the structure to reflect your CL (Ifindex, Type, etc...) and also switched to using a const for Sizeofpacket_mreq instead of unsafe.Sizeof().

In the interest of not inducing compile conflict when /x/sys/unix gets updated I've kept packet_mreq and Sizeofpacket_mreq naming convention.